tests/pulse/test_parametrized_hamiltonian.py

```diff
@@ -124,8 +129,18 @@ def test_H_parametrized(self):
 
     def test__repr__(self):
         """Test repr method returns expected string"""
-        str = repr(test_example)
-        assert str == "ParametrizedHamiltonian: terms=4"
+        coeffs = [2.0, f1, f2]
+        ops = [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+        H = ParametrizedHamiltonian(coeffs, ops)
+        expected = "(2.0*(PauliX(wires=[0])))+(f1(params_0,t)*(PauliY(wires=[0])))+(f2(params_1,t)*(PauliZ(wires=[0])))"
+
+        assert repr(H).replace("\n", "").replace(" ", "") == expected
+
+        coeffs = [2.0, f1, f2, f3(0.5)]
+        observables = [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0), qml.PauliX(0)]
+        H = ParametrizedHamiltonian(coeffs, observables)
+        expected = "(2.0*(PauliX(wires=[0])))+(f1(params_0,t)*(PauliY(wires=[0])))+(f2(params_1,t)*(PauliZ(wires=[0])))+(f3(params_2,t)*(PauliX(wires=[0])))"
+        assert repr(H).replace("\n", "").replace(" ", "") == expected
```

Comment:
Sorry for being pedantic, but can you please separate this out into a second test. Something like `test_repr_with_class_objects`

Reply:
Certainly